descope/mgmt/sdk.go

```diff
@@ -0,0 +1,102 @@
+package mgmt
+
+// Provides functions for managing tenants in a project.
+type Tenant interface {
+	// Create a new tenant with the given name.
+	//
+	// selfProvisioningDomains is an optional list of domains that are associated with this
+	// tenant. Users authenticating from these domains will be associated with this tenant.
+	//
+	// The tenant name must be unique per project. The tenant ID is generated automatically
+	// for the tenant.
+	Create(managementKey, name string, selfProvisioningDomains []string) (id string, err error)
+
+	// Create a new tenant with the given name and ID.
+	//
+	// selfProvisioningDomains is an optional list of domains that are associated with this
+	// tenant. Users authenticating from these domains will be associated with this tenant.
+	//
+	// Both the name and ID must be unique per project.
+	CreateWithID(managementKey, id, name string, selfProvisioningDomains []string) error
+
+	// Update an existing tenant's name and domains.
+	//
+	// IMPORTANT: All parameters are required and will override whatever value is currently
+	// set in the existing tenant. Use carefully.
+	Update(managementKey, id, name string, selfProvisioningDomains []string) error
+
+	// Delete an existing tenant.
+	//
+	// IMPORTANT: This action is irreversible. Use carefully.
+	Delete(managementKey, id string) error
+}
+
+// Represents a tenant association for a User. The tenant ID is required to denote
+// which tenant the user belongs to. Roles is an optional list of roles for the
+// user in this specific tenant.
+type UserTenants struct {
+	TenantID string
+	Roles    []string
+}
+
+// Provides functions for managing users in a project.
+type User interface {
+	// Create a new user.
+	//
+	// The identifier is required and will determine what the user will use to
+	// sign in. All other fields are optional.
+	//
+	// The roles parameter is an optional list of the user's roles for users that
+	// aren't associated with a tenant, while the tenants parameter can be used
+	// to specify which tenants to associate the user with and what roles the
+	// user has in each one.
+	Create(managementKey, identifier, email, phone, displayName string, roles []string, tenants []UserTenants) error
+
+	// Update an existing user.
+	//
+	// The parameters follow the same convention as those for the Create function.
+	//
+	// IMPORTANT: All parameters will override whatever values are currently set
+	// in the existing user. Use carefully.
+	Update(managementKey, identifier, email, phone, displayName string, roles []string, tenants []UserTenants) error
+
+	// Delete an existing user.
+	//
+	// IMPORTANT: This action is irreversible. Use carefully.
+	Delete(managementKey, identifier string) error
+}
+
+// Represents a mapping between a set of groups of users and a role that will be assigned to them.
+type RoleMapping struct {
```

Comment:
Not critical
But i think that we placed the structs in a different place in auth sdk